include/zephyr/net/net_ip.h

```diff
@@ -745,6 +745,39 @@ static inline bool net_ipv6_is_prefix(const uint8_t *addr1,
 	return (addr1[bytes] & mask) == (addr2[bytes] & mask);
 }
 
+
+/**
+ * @brief Get the IPv6 network address via the unicast address and the prefix mask.
+ *
+ * @param inaddr Unicast IPv6 address.
+ * @param outaddr Prefix masked IPv6 address (network address).
+ * @param length Prefix length (max length is 128).
+ */
```

Comment:
Please place this function in a separate commit (we do not want to mix sample changes to other network changes in the same commit).